packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseComponentsObject-test.js

```diff
@@ -322,6 +322,24 @@ describe('Components Object', () => {
       expect(securitySchemes.get(0)).to.be.instanceof(namespace.elements.AuthScheme);
       expect(securitySchemes.get(0).meta.id.toValue()).to.equal('token');
     });
+
+    it('handles invalid security scheme', () => {
+      const components = new namespace.elements.Object({
+        securitySchemes: {
+          Basic: null,
+        },
+      });
+
+      const parseResult = parse(context, components);
+      expect(parseResult.length).to.equal(2);
+
+      const parsedComponents = parseResult.get(0);
+      expect(parsedComponents).to.be.instanceof(namespace.elements.Object);
+
+      const schemes = parsedComponents.get('securitySchemes');
+      expect(schemes).to.be.instanceof(namespace.elements.Array);
```

Comment:
@pksunkara I don't think we should return an array here but instead an object like the other components parsers. This is inconsistent with all other component parsers and means we cannot use generic functions like `parseReference` as it relies on an object being here.

Outside scope of this fix though but will be needed to handle dereferencing these values.

Reply:
I think the fix should be part of this PR or it should wait until both this and the main security PR is merged which might take some time. IMHO, should be part of this PR.

Reply:
You mentioned holding off on merging this to prevent a complicated merge/rebase for #206. I don't think that will be a problem, #206 does not touch parseComponentsObject or related tests.